visualization/app/codeCharta/ui/mapTreeViewSearch/mapTreeViewSearch.component.spec.ts

```diff
@@ -95,17 +95,84 @@ describe("MapTreeViewSearchController", () => {
+        const blacklistItem = {path: "/root/node/path", type: BlacklistType.exclude};
+        const anotherBlacklistItem = {path: "/root/another/node/path", type: BlacklistType.exclude};
+        mapTreeViewSearchController.settingsService.settings.blacklist = [blacklistItem, anotherBlacklistItem];
+        mapTreeViewSearchController.viewModel.searchPattern = "/root/node/path";
+
+        mapTreeViewSearchController.updateViewModel();
+        expect(mapTreeViewSearchController.viewModel.isPatternHidden).toBeFalsy();
+        expect(mapTreeViewSearchController.viewModel.isPatternExcluded).toBeTruthy();
+    });
+
+    it("should updateViewModel when pattern hidden and excluded", () => {
+        const blacklistItemExcluded = {path: "/root/node/path", type: BlacklistType.exclude};
+        const blacklistItemHidden = {path: "/root/node/path", type: BlacklistType.hide};
+        mapTreeViewSearchController.settingsService.settings.blacklist = [blacklistItemExcluded, blacklistItemHidden];
+        mapTreeViewSearchController.viewModel.searchPattern = "/root/node/path";
+
+        mapTreeViewSearchController.updateViewModel();
+        expect(mapTreeViewSearchController.viewModel.isPatternHidden).toBeTruthy();
+        expect(mapTreeViewSearchController.viewModel.isPatternExcluded).toBeTruthy();
+    });
+
+
+    describe("Update ViewModel with blacklist count", () => {
+
+        let file1, file2, folder1;
+
+        beforeEach(() => {
+            CodeMapUtilService.getNodesByGitignorePath = jest.fn(() => {
+                return [file1, file2, folder1];
+            });
+
+            file1 = simpleHierarchy.children[0].children[0].children[0];
+            file2 = simpleHierarchy.children[0].children[1];
+            folder1 = simpleHierarchy.children[0].children[0];
         });
-        mapTreeViewSearchController.settingsService.applySettings = jest.fn();
 
-        mapTreeViewSearchController.setSearchedNodePathnames();
+        it("should get correct searchedNodePaths with searchedFiles.length", () => {
+            mapTreeViewSearchController.settingsService.applySettings = jest.fn();
 
-        expect(mapTreeViewSearchController.settingsService.settings.searchedNodePaths).toEqual(["/root/a", "/root/a/ab"]);
-        expect(mapTreeViewSearchController.viewModel.folderCount).toBe(2);
-        expect(mapTreeViewSearchController.viewModel.fileCount).toBe(0);
-        expect(mapTreeViewSearchController.settingsService.applySettings).toHaveBeenCalled();
+            mapTreeViewSearchController.setSearchedNodePathnames();
+
+            expect(mapTreeViewSearchController.settingsService.settings.searchedNodePaths).toEqual(["/root/a/ab/aba", "/root/a/abc", "/root/a/ab"]);
+            expect(mapTreeViewSearchController.searchedFiles.length).toBe(2);
+            expect(mapTreeViewSearchController.settingsService.applySettings).toHaveBeenCalled();
+        });
+
+        xit("should get correct fileCount, hideCOunt and excludeCount", () => {
```

Comment:
@alschmut I resolved a bigger merge conflict. I then realized, that this this test fails. I tried running it on the unmerged branch and it still failed. Can you take a look into it ?

Reply:
@mrpatpat Alright I gonna talk e a look (:

Reply:
@mrpatpat actually it worked for me.. And TravisCI seems to have no problem either